lib/engine/game/g_18_dixie/step/icgscl_merge_consent.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+require_relative '../../../step/base'
+
+module Engine
+  module Game
+    module G18Dixie
+      module Step
+        class MergeConsent < Engine::Step::Base
+          def round_state
+            {
+              merge_consent_merging_corp: nil,
+              merge_consent_primary_corp: nil,
+              merge_consent_secondary_corp: nil,
+              merge_consent_pending_corps: [],
+              merge_consent_subsidy: nil,
+              merge_presidency_cash_crisis_corp: nil,
+              merge_presidency_cash_crisis_player: nil,
+              merge_presidency_exchange_corps: [],
+              merge_presidency_exchange_merging_corp: nil,
+              merge_presidency_exchange_subsidy: nil,
+            }
+          end
+
+          def merger_corp_name
+            @round.merge_consent_merging_corp&.name || 'SCL/ICG'
+          end
+
+          def active_corp_name
+            @round.merge_consent_pending_corps[0]&.name || 'current corporation'
+          end
+
+          def primary_corp_name
+            @round.merge_consent_primary_corp&.name || 'first forming corporation'
+          end
+
+          def secondary_corp_name
+            @round.merge_consent_secondary_corp&.name || 'second forming corporation'
+          end
+
+          def secondary_corp_president
+            @round.merge_consent_pending_corps[1]&.owner&.name || 'other merging corporation\'s president'
+          end
+
+          def merge_description
+            "merge #{primary_corp_name} and #{secondary_corp_name} to form the #{merger_corp_name}"
+          end
+
+          def active_entities
+            @round.merge_consent_pending_corps
+          end
+
+          def active?
+            !@round.merge_consent_pending_corps.empty?
+          end
+
+          def actions(entity)
+            return %w[choose] if turn_to_choose(entity)
+
+            []
+          end
+
+          def show_other
+            [@round.merge_consent_primary_corp,
+             @round.merge_consent_secondary_corp,
+             @round.merge_consent_merging_corp].freeze - [current_entity]
```

Comment:
what's the point of the freeze here?

Reply:
When I looked up the behavior of "array - array" the site I used had me think that the first array may be mutated. I've removed the `.freeze`